### ISSUE TITLE
/04_operators lack of logical operators

### ISSUE THREAD

**Issue by BroBQ** [Xeverous/Xeverous.github.io#20]
I'm sure at this chapter logical operators should be present
```
!a
a && b
a || b
```
and also `and` and `or`
https://en.cppreference.com/w/cpp/language/operator_alternative

**Comment by Xeverous** [comment]
This is actually a complex tutorial ordering problem because:

- logical operators do not offer much before introduction of control flow statements. It's hard to give any real-life example before `if`
- logical operators are often used with short-circuit evaluation - the best examples are with pointers... 

This needs either to be split across different chapters or present some forward magical examples in operator chapter.

What do you propose?

**Comment by BroBQ** [comment]
I suggest Cisco attempt, first introduction, then more information when necessary. We would let people know something like it exists without explaining much about it.

First truth tables should be introduced for AND, OR, and NOT.

I think analogy to logic gates should be used. We know that CPU has billions of transistors but two of them are creating a logic gate. In this way CPU can compare two inputs (let's say 5 V and 0 V; HIGH state and LOW state, TRUE and FALSE) and as output it will send other value (value wanted by circuit designer).

Then add spoiler with pictures of actual scheme of transistor gates and picture of **little** bigger brother (SN7400N) of transistors located at CPU. Also, please notice George Boole who did this without knowledge about existence of computers or transistors.


explaining code:
```
#include <iostream>

int main()
{
    bool a = true;
    bool b = true;
    bool c = a && b; // true and true = true
    bool d = a || b; // true or true = true
    bool e = !a && b; // (not)true and true; means false and true = false
    bool f = !a || b; // (not)true true true; means false or true = true
    std::cout << std::boolalpha;
    std::cout << "true and true = " << c << std::endl;
    std::cout << "true or true = " << d << std::endl;
    std::cout << "false or true = " << e << std::endl;
    std::cout << "false and true = " << f << std::endl;
    return 0;
}
```
What's more, in this code any magical force does not appear, everything should be well known. No exercises, just some info.
Exercises with logical operator should be at if article

**Comment by Xeverous** [comment]
This is a good sugesstion. I already wanted to place somewhere videos about logic gates (constructed using domino!) and this seems to be a good moment.

**Comment by Xeverous** [comment]
Fixed in bc8ff4646d108a4006f1a8bed0207fa13f19e486.
